lib/collection/src/collection/snapshots.rs

```diff
@@ -184,12 +184,9 @@ impl Collection {
                 if !mapping_path.exists() {
                     Vec::new()
                 } else {
-                    let shard_key_mapping: ShardKeyMapping = read_json(&mapping_path)?;
-                    shard_key_mapping
-                        .values()
-                        .flat_map(|v| v.iter())
-                        .copied()
-                        .collect()
+                    // Use wrapper type to support both formats
+                    let shard_key_mapping: ShardKeyMappingWrapper = read_json(&mapping_path)?;
+                    shard_key_mapping.shard_ids()
```

Comment:
This is where we failed to recover the new shard format, which resulted in the CI errors [here](https://github.com/qdrant/qdrant/pull/6194#pullrequestreview-2696112315).